### ISSUE TITLE
Overflowing multiplication in lrf.rs:sgrproj_box_sum_slow in 420/12 bit

### ISSUE THREAD

**Issue by sdroege** [xiph/rav1e#1300]
This only happens in 12 bit 420 mode. 8/10 bit modes all work, 422/444 also work in 12 bit more. All on the same 320x240 input, but the input does not seem to matter.

```
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:381
   6: rust_begin_unwind
             at src/libstd/panicking.rs:308
   7: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
   8: core::panicking::panic
             at src/libcore/panicking.rs:49
   9: rav1e::lrf::sgrproj_box_sum_slow
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/lrf.rs:85
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/lrf.rs:179
  10: rav1e::lrf::sgrproj_box_ab_r2
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/lrf.rs:409
  11: rav1e::lrf::sgrproj_solve
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/lrf.rs:589
  12: rav1e::rdo::rdo_loop_decision
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/rdo.rs:1399
  13: rav1e::encoder::encode_tile
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/encoder.rs:2193
  14: rav1e::encoder::encode_tile_group::{{closure}}
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/encoder.rs:2008
  15: <rayon::iter::map::MapFolder<C,F> as rayon::iter::plumbing::Folder<T>>::consume
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/map.rs:225
  16: rayon::iter::plumbing::Folder::consume_iter
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/plumbing/mod.rs:176
  17: rayon::iter::plumbing::Producer::fold_with
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/plumbing/mod.rs:108
  18: rayon::iter::plumbing::bridge_producer_consumer::helper
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/plumbing/mod.rs:418
  19: rayon::iter::plumbing::bridge_producer_consumer
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/plumbing/mod.rs:391
  20: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/plumbing/mod.rs:368
  21: <rayon::vec::IntoIter<T> as rayon::iter::IndexedParallelIterator>::with_producer
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/vec.rs:66
  22: rayon::iter::plumbing::bridge
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/plumbing/mod.rs:351
  23: <rayon::vec::IntoIter<T> as rayon::iter::ParallelIterator>::drive_unindexed
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/vec.rs:33
  24: <rayon::iter::map::Map<I,F> as rayon::iter::ParallelIterator>::drive_unindexed
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/map.rs:52
  25: <rayon::iter::unzip::UnzipB<I,OP,CA> as rayon::iter::ParallelIterator>::drive_unindexed
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/unzip.rs:263
  26: rayon::iter::collect::special_extend
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/collect/mod.rs:41
  27: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/collect/mod.rs:132
  28: <rayon::iter::unzip::UnzipA<I,OP,FromB> as rayon::iter::ParallelIterator>::drive_unindexed
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/unzip.rs:216
  29: rayon::iter::collect::special_extend
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/collect/mod.rs:41
  30: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/collect/mod.rs:132
  31: rayon::iter::unzip::execute_into
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/unzip.rs:56
  32: rayon::iter::unzip::execute
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/unzip.rs:36
  33: rayon::iter::unzip::unzip
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/unzip.rs:71
  34: rayon::iter::ParallelIterator::unzip
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-1.0.3/src/iter/mod.rs:1659
  35: rav1e::encoder::encode_tile_group
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/encoder.rs:2002
  36: rav1e::encoder::encode_frame
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/encoder.rs:2275
  37: rav1e::api::ContextInner<T>::receive_packet
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/api.rs:875
  38: rav1e::api::Context<T>::receive_packet::{{closure}}
             at /home/slomo/.cargo/git/checkouts/rav1e-14e16d25dd5cd9c0/a38059c/src/api.rs:606
  39: rayon_core::thread_pool::ThreadPool::install::{{closure}}
             at /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/rayon-core-1.4.1/src/thread_pool/mod.rs:124
```

**Comment by lu-zero** [comment]
Could you please provide also a sample to reproduce the issue?

**Comment by sdroege** [comment]
Can be reproduced with the GStreamer rav1e plugin from: https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/tree/master/gst-plugin-rav1e

```sh
RUST_BACKTRACE=1 gst-launch-1.0 videotestsrc ! "video/x-raw,format=I420_12LE" ! rav1enc ! fakesink
```

**Comment by sdroege** [comment]
[test.y4m.zip](https://github.com/xiph/rav1e/files/3219413/test.y4m.zip)

This reproduces the panic with the `rav1e` binary if called with no special parameters.

**Comment by tmatth** [comment]
This clip also desyncs, and the underlying issue predates the regression in #1286.

**Comment by tmatth** [comment]
Fixed in #1305

**Comment by sdroege** [comment]
Thanks!
